neuro_san/internals/chat/data_driven_chat_session.py

```diff
@@ -135,9 +135,10 @@ async def chat(self, user_input: str,
             logger = logging.getLogger(self.__class__.__name__)
             logger.error(traceback.format_exc())
 
+        converter = BaseMessageDictionaryConverter(origin=self.front_man.get_origin())
         chat_messages: List[Dict[str, Any]] = []
         for raw_message in raw_messages:
-            chat_message: Dict[str, Any] = convert_to_chat_message(raw_message, self.front_man.get_origin())
+            chat_message: Dict[str, Any] = converter.to_dict(raw_message)
```

Comment:
Use the BaseMessageDictionaryConverter factored out from message_utils methods to do the conversion.